modules/view/main/index.coffee

```diff
@@ -68,14 +68,26 @@ class View
           for j in [i..data.length-1]
             enterSet.push {datum: data[j]}
 
+      viewEnterWarning = (element, selector) ->
+        hx.consoleWarning "view enter fn returned", element, "! It didn't match selector", selector, ", so you may encounter odd behavior"
 
       newNodeSet = enterSet.map (d, i) =>
         datum = d.datum
         element = @new.call @rootSelection, d.datum, i
-        subselection = @rootSelection.selectAll @selector
 
-        unless element in subselection.nodes
-          hx.consoleWarning "view enter fn returned", element, "! It didn't match selector", @selector, ", so you may encounter odd behavior"
+        if @selector.indexOf('.') >= 0
+          classes = @selector.split('.')
+          isChild = @rootSelection.node().contains(element)
+
+          # Checks isChild first as it's the quickest operation
+          if not isChild
+            viewEnterWarning(element, @selector)
+          else
+            isClassedCorrectly = hx.select(element).classed(classes.slice(1).join(' '))
```

Comment:
The first item in the array is either an empty string or the element type of the element (i.e. for `div.class` it would be `div` whereas for `.class` it would be empty string). We also care about all the classes in the selector so we check them all here using `.classed`